crates/manifest/src/schema/v2.rs

```diff
@@ -145,11 +174,116 @@ pub struct Component {
     /// Settings for custom tools or plugins. Spin ignores this field.
     #[serde(default, skip_serializing_if = "Map::is_empty")]
     pub tool: Map<String, toml::Table>,
+    /// If true, allow dependencies to inherit configuration.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub dependencies_inherit_configuration: Option<bool>,
+    /// Component dependencies
+    #[serde(default, skip_serializing_if = "ComponentDependencies::is_empty")]
+    pub dependencies: ComponentDependencies,
+}
+
+/// Component dependencies
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(try_from = "Map<DependencyName, ComponentDependency>")]
+pub struct ComponentDependencies {
+    /// `dependencies = { "foo:bar" = ">= 0.1.0" }`
+    pub inner: Map<DependencyName, ComponentDependency>,
+}
+
+impl TryFrom<Map<DependencyName, ComponentDependency>> for ComponentDependencies {
+    type Error = anyhow::Error;
+
+    fn try_from(value: Map<DependencyName, ComponentDependency>) -> Result<Self, Self::Error> {
+        let dependencies = ComponentDependencies { inner: value };
+        dependencies.validate()?;
+        Ok(dependencies)
+    }
+}
+
+impl ComponentDependencies {
+    fn validate(&self) -> anyhow::Result<()> {
```

Comment:
This feels like a lot of logic to have in a schema - this puts us into "we can't even deserialise the thing to reason about it" territory.  For some reason I thought we had a separate "validate" step but maybe not.  And maybe this is no different from the way we fail the read for kebab-id violations.  Anyway, just sharing it as a reservation...!

Reply:
Do you have reservation about this code living in `v2.rs` or just about from where `validate` is called? I agree that it makes sense to call `validate` from the loader for the reasons you stated.

Reply:
Updated to defer validating dependency sections until locking (just after normalization). PTAL, thanks!